olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -264,6 +274,24 @@ AuthenticationClient::Impl::Impl(const std::string& authentication_server_url,
               token_cache_limit)),
       network_settings_() {}
 
+AuthenticationClient::Impl::Impl(AuthenticationSettings settings,
+                                 const std::string& authentication_server_url,
+                                 size_t token_cache_limit)
+    : server_url_(authentication_server_url),
+      client_token_cache_(
+          std::make_shared<Atomic<utils::LruCache<std::string, SignInResult>>>(
+              token_cache_limit)),
+      user_token_cache_(std::make_shared<
+                        Atomic<utils::LruCache<std::string, SignInUserResult>>>(
+          token_cache_limit)),
+      network_settings_() {
+  if (settings.network_proxy_settings) {
+    SetNetworkProxySettings(settings.network_proxy_settings.get());
+  }
+  SetNetwork(settings.network_request_handler);
+  SetTaskScheduler(settings.task_scheduler);
```

Comment:
I would rather do it the other way around.
Store a copy of AuthenticationSettings internally and the setters modify that member values.
Else we need to touch it again when we remove the setters.